server/conn.go

```diff
@@ -1066,6 +1068,19 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	}
 }
 
+func (cc *clientConn) writeStats(ctx context.Context) error {
+	msg := []byte("Uptime: 0  Threads: 0  Questions: 0  Slow queries: 0  Opens: 0  Flush tables: 0  Open tables: 0  Queries per second avg: 0.000")
```

Comment:
I could modify it like this:
```go
$ git diff -- server/conn.go util/sys/linux/sys_linux.go
diff --git a/server/conn.go b/server/conn.go
index 698571dfd..0f1c50631 100644
--- a/server/conn.go
+++ b/server/conn.go
@@ -81,6 +81,7 @@ import (
        "github.com/pingcap/tidb/util/logutil"
        "github.com/pingcap/tidb/util/memory"
        "github.com/pingcap/tidb/util/sqlexec"
+       "github.com/pingcap/tidb/util/sys/linux"
        "github.com/prometheus/client_golang/prometheus"
        "go.uber.org/zap"
 )
@@ -1069,7 +1070,11 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 }
 
 func (cc *clientConn) writeStats(ctx context.Context) error {
-       msg := []byte("Uptime: 0  Threads: 0  Questions: 0  Slow queries: 0  Opens: 0  Flush tables: 0  Open tables: 0  Queries per second avg: 0.000")
+       // Note that `mysqlsh` (MySQL Shell) parses out the `Uptime` when running `\status`.
+       // for this to work this needs to be an integer with one space before and two spaces
+       // after it.
+       stat := fmt.Sprintf("Uptime: %d  ", linux.Uptime())
+       msg := []byte(stat)
        data := cc.alloc.AllocWithLen(4, len(msg))
        data = append(data, msg...)
 
diff --git a/util/sys/linux/sys_linux.go b/util/sys/linux/sys_linux.go
index 976815956..f361eb3f9 100644
--- a/util/sys/linux/sys_linux.go
+++ b/util/sys/linux/sys_linux.go
@@ -52,3 +52,9 @@ func SetAffinity(cpus []int) error {
        }
        return unix.SchedSetaffinity(unix.Getpid(), &cpuSet)
 }
+
+func Uptime() int64 {
+       si := &unix.Sysinfo_t{}
+       unix.Sysinfo(si)
+       return si.Uptime
+}
```
This would return the system (not TiDB) uptime of the host running this process. This might not be the best thing, but it is more informative than always returning 0. Removing the other values seem to work just fine, at least for `mysqlsh`.

This seems to be slightly related to https://github.com/pingcap/tidb/issues/8842 as that should probably return the same value for consistency.

Reply:
Yeah, I prefer this - but lets see what other reviewers prefer too. I commented in #8842 that it is possible to retrieve the correct uptime from prometheus, but because it is an optional component it might not be a good fit here.